tests/providers/aws/test_s3.py

```diff
@@ -0,0 +1,63 @@
+#
+# Copyright 2018 Red Hat, Inc.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as
+# published by the Free Software Foundation, either version 3 of the
+# License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+#
+
+"""Test the AWS S3 utility functions."""
+
+import os.path
+import random
+import tempfile
+from unittest.mock import ANY, Mock, patch, PropertyMock
+
+from faker import Faker
```

Comment:
I'd like to see us using Moto for AWS tests - https://github.com/spulec/moto

I won't make you rewrite this right now, but I would like to see future PRs avoid reinventing this wheel. :smiley: